phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -4134,25 +4151,34 @@ public MutationState addColumn(PTable table, List<ColumnDef> origColumnDefs,
                                 metaPropertiesEvaluated.getUpdateCacheFrequency(),
                                 metaPropertiesEvaluated.getPhoenixTTL(),
                                 metaPropertiesEvaluated.getPhysicalTableName(),
-                                metaPropertiesEvaluated.getSchemaVersion());
+                                metaPropertiesEvaluated.getSchemaVersion(),
+                                metaPropertiesEvaluated.getColumnEncodedBytes());
                     }
                     tableMetaData.addAll(connection.getMutationState().toMutations(timeStamp).next().getSecond());
                     connection.rollback();
                 }
 
+                long seqNum = 0;
                 if (changingPhoenixTableProperty || columnDefs.size() > 0) {
-                    incrementTableSeqNum(table, tableType, columnDefs.size(), metaPropertiesEvaluated);
+                    seqNum = incrementTableSeqNum(table, tableType, columnDefs.size(), metaPropertiesEvaluated);
 
                     tableMetaData.addAll(connection.getMutationState().toMutations(timeStamp).next().getSecond());
                     connection.rollback();
                 }
 
+               if (isTransformNeeded) {
+                   try {
+                       Transform.addTransform(connection, tenantIdToUse, table, metaProperties, seqNum, PTable.TransformType.METADATA_TRANSFORM);
+                    } catch (SQLException ex) {
+                       connection.rollback();
+                       throw ex;
```

Comment:
Looks like this can happen only if we fail to parse metaProperties object to String, correct?